app/src/main/java/co/temy/android/ktx/view/ViewExtensions.kt

```diff
@@ -0,0 +1,558 @@
+ */
+
+private fun LinearLayoutManager.getCurrentPositionForVerticalOrientation(midScreenY: Int): Int {
+    if (this.orientation == LinearLayoutManager.VERTICAL) {
+        val centerPosition = getCenterPositionForVerticalOrientation(midScreenY)
+        return if (centerPosition == RecyclerView.NO_POSITION) {
+            getCompletelyVisibleOrFirstPosition()
+        } else centerPosition
+    }
+
+    return RecyclerView.NO_POSITION
+}
+
+/**
+ * Get position of center item for [LinearLayoutManager.VERTICAL]
+ *
+ * @return position of item with start Y less than middle of the screen and end Y more than middle of the screen
+ *
+ * @return [RecyclerView.NO_POSITION] if [LinearLayoutManager] doesn't have vertical orientation or there is no item that meets requirements
+ */
+private fun LinearLayoutManager.getCenterPositionForVerticalOrientation(midScreenY: Int) : Int {
+    if (this.orientation == LinearLayoutManager.VERTICAL) {
+        val firstPosition = this.findFirstVisibleItemPosition()
+        val lastPosition = this.findLastVisibleItemPosition()
+
+        for (position in firstPosition..lastPosition) {
+            val view = this.findViewByPosition(position)
+            if (view != null) {
+                val viewHeight = view.measuredHeight
+                val viewStartY = view.y
+                val viewEndY= viewStartY + viewHeight
+                if (viewStartY < midScreenY && viewEndY > midScreenY) {
+                    return position
+                }
+            }
+        }
+
+        return RecyclerView.NO_POSITION
+    }
+
+    return RecyclerView.NO_POSITION
+}
+
+fun View.setBottomPadding(bottomPadding: Int) {
+    setPadding(paddingLeft, paddingTop, paddingRight, bottomPadding)
+}
+
+fun Context.inflate(resource: Int, root: ViewGroup? = null, attachToRoot: Boolean = false): View {
+    return LayoutInflater.from(this).inflate(resource, root, attachToRoot)
+}
+
+fun Context.getColorCompat(@ColorRes color: Int) = ContextCompat.getColor(this, color)
+
+fun Context.getDrawableCompat(@DrawableRes drawable: Int) = ContextCompat.getDrawable(this, drawable)
+
+/**
+ * Triggers when recycler view state changes to [RecyclerView.SCROLL_STATE_IDLE].
+ */
+inline fun RecyclerView.addOnIdleStateListener(crossinline listener: (RecyclerView) -> Unit) {
+    addOnScrollListener(object : RecyclerView.OnScrollListener() {
+        override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+            if (newState == RecyclerView.SCROLL_STATE_IDLE) {
+                listener.invoke(recyclerView)
+            }
+        }
+    })
+}
+
+/**
+ * Converts value in pixels into value in device-independent pixels
+ */
+fun Int.toDp(): Int {
+    val metrics = Resources.getSystem().displayMetrics
+    return this / (metrics.densityDpi / DisplayMetrics.DENSITY_DEFAULT)
+}
+
+/**
+ * Converts value in device-independent pixels into value in pixels
+ */
+fun Int.toPx(): Int {
```

Comment:
Also, a question, if this should be a part of View extensions, or maybe Resource, Contex ? ..